simSPI/fov.py

```diff
@@ -0,0 +1,249 @@
+    detector_ly = detector_ny_pix * detector_pix_size
+    fov_lx = detector_lx / magnification
+    fov_ly = detector_ly / magnification
+
+    if pdb_file is not None:
+        dmax = get_dmax(pdb_file)
+
+    boxsize = dmax + 2 * pad
+    return fov_lx, fov_ly, boxsize
+
+
+def get_dmax(filename):
+    """Get maximmum dimension of particle.
+
+    Parameters
+    ----------
+    filename : str
+        Relative path to file containing topological information of particle
+
+    Returns
+    -------
+    np.amax(distance) : float
+        Max dimension of given particle in .pdb source file.
+    """
+    xyz = get_xyz_from_pdb(filename)
+    distance = pdist(xyz[0, ...])
+    dmax = np.amax(distance)
+
+    return dmax
+
+
+def get_xyz_from_pdb(filename=None, atom_selection="name CA or name P"):
+    """Get particle coordinates from .pdb file.
+
+    Parameters
+    ----------
+    filename : str
+        Relative path to file containing topological information of particle
+    atom_selection: str
+        Atoms to be selected by MDTraj for returning their cartesian coordinates
+
+    Returns
+    -------
+    traj_small.xyz : ndarray
+        Particle coordinates from .pdb file.
+    """
+    traj = md.load(filename)
+    atom_indices = traj.topology.select(atom_selection)
+    traj_small = traj.atom_slice(atom_indices)
+    coordinates = traj_small.xyz
+
+    return coordinates
+
+
+def micrograph2particles(
+    micrograph, optics_params, detector_params, pdb_file=None, dmax=30, pad=5.0
+):
+    """Extract particles from given micrograph.
+
+    Parameters
+    ----------
+    micrograph : ndarray
+        Particle micrograph to extract from.
+    optics_params : list
+        List of sim parameters pertaining to microscope settings.
+    detector_params : list
+        List of sim parameters pertaining to detector settings.
+    pdb_file : str
+        Relative path to .pdb file.
+    dmax : int
+        Predefined maximum dimension of molecule.
+    pad : int
+        Amount of padding for each particle box.
+
+    Returns
+    -------
+    particles : ndarray
+        Picked and sliced particle data from micrograph.
+    """
+    fov_Lx, fov_Ly, boxsize = get_fov(
+        optics_params, detector_params, pdb_file=pdb_file, dmax=dmax, pad=pad
+    )
+    fov_Nx = np.floor(fov_Lx / boxsize)
+    fov_Ny = np.floor(fov_Ly / boxsize)
+    pixel_size = (fov_Lx / micrograph.shape[1] + fov_Ly / micrograph.shape[0]) / 2.0
+    n_boxsize = np.int(boxsize / pixel_size)
+    x_pixels = np.int(fov_Nx * n_boxsize)
+    y_pixels = np.int(fov_Ny * n_boxsize)
+    data = micrograph[0:y_pixels, 0:x_pixels]
+    particles = slice_and_stack(data, n_boxsize=n_boxsize)
+
+    return particles
+
+
+def slice_and_stack(data, n_boxsize=256, n_ovl=0):
+    """Convert a 2D numpy array into a 3D numpy array.
+
+    Parameters
+    ----------
+    data : ndarray
+        2D array to stack.
+    n_boxsize : int
+        Boxsize.
+    n_ovl : int
+        Overlap.
```

Comment:
What does overlap means for this function? Could you explain a bit in the docstring?

Reply:
to be honest i am not sure - @fredericpoitevin maybe you can elaborate more? that else statement is never executed because `n_ovl` is always 0, so maybe we can just remove that part?